src/pb.rs

```diff
@@ -322,38 +322,43 @@ impl<T: Write> ProgressBar<T> {
         let speed = self.current as f64 / fract_dur(time_elapsed);
         let width = self.width();
 
+        let mut out;
+        let mut parts = Vec::new();
         let mut base = String::new();
-        let mut suffix = String::new();
         let mut prefix = String::new();
-        let mut out;
+        let mut suffix = String::from(" ");
 
         // precent box
         if self.show_percent {
             let percent = self.current as f64 / (self.total as f64 / 100f64);
-            suffix =
-                suffix + &format!(" {:.*} % ", 2, if percent.is_nan() { 0.0 } else { percent });
+            parts.push(format!(
+                "{:.*} %",
+                2,
+                if percent.is_nan() { 0.0 } else { percent }
+            ));
         }
         // speed box
         if self.show_speed {
-            suffix = match self.units {
-                Units::Default => suffix + &format!("{:.*}/s ", 2, speed),
-                Units::Bytes => suffix + &format!("{}/s ", kb_fmt!(speed)),
+            match self.units {
+                Units::Default => parts.push(format!("{:.*}/s", 2, speed)),
+                Units::Bytes => parts.push(format!("{}/s", kb_fmt!(speed))),
             };
         }
         // time left box
-        if self.show_time_left && self.current > 0 {
-            if self.total > self.current {
-                let left = 1. / speed * (self.total - self.current) as f64;
-                if left < 60. {
-                    suffix = suffix + &format!("{:.0}s", left);
-                } else {
-                    suffix = suffix + &format!("{:.0}m", left / 60.);
-                }
-            }
+        if self.show_time_left && self.current > 0 && self.total > self.current {
+            let left = 1. / speed * (self.total - self.current) as f64;
+            if left < 60. {
+                parts.push(format!("{:.0}s", left));
+            } else {
+                parts.push(format!("{:.0}m", left / 60.));
+            };
+        }
+        if parts.len() > 0 {
```

Comment:
I believe it is recommended to use `is_empty()` instead of `len() > 0`. However, I think you could just drop the condition altogether since joining an empty vec will  produce an empty string anyways.